### DISCUSSION TITLE
How can I get a subset of a ggml tensor?

### DISCUSSION THREAD

**Discussion by mecccaaa** [ggml-org/ggml#1055] [Q&A]
For example, I have a 2D ggml tensor shaped as [B,T], how can I get the subset tensor which removed the last column and shaped as [B, T-1]? Thanks!

**Comment by ggerganov** [comment]
Use `ggml_view_2d`

**Reply by mecccaaa** [reply]
I tried, but using ggml_view_2d actually just removed the last B elements of the entire tensor, not the last column.
